core/src/main/java/org/apache/iceberg/deletes/BitmapPositionDeleteIndex.java

```diff
@@ -43,6 +53,11 @@ class BitmapPositionDeleteIndex implements PositionDeleteIndex {
     this.deleteFiles = deleteFile != null ? Lists.newArrayList(deleteFile) : Lists.newArrayList();
   }
 
+  BitmapPositionDeleteIndex(RoaringPositionBitmap bitmap, DeleteFile deleteFile) {
+    this.bitmap = bitmap;
+    this.deleteFiles = deleteFile != null ? Lists.newArrayList(deleteFile) : Lists.newArrayList();
```

Comment:
Is there a case where we can have a deserialized bitmap, but no delete file?  Just wondering if we should have a separate constructor or if it makes sense to allow passing null for those cases.

Reply:
I don't think so. I have this check here to match the constructor above. There are some legacy code paths that create an index and pass null as `DeleteFile`.